### ISSUE TITLE
Something went wrong. cron reached maximum iterations

### ISSUE THREAD

**Issue by fabnavigator** [kelektiv/node-cron#408]
Time Zone: "" - Cron String: 0 0 2 * * * - UTC offset: -04:00 - current Date: Sat Mar 09 2019 08:32:59 GMT-0500

**Comment by GaneshaThirumurthi** [comment]
Having the same problem. Below is my error log. 

@fabnavigator what version are you running? I am on 1.5

/app/node_modules/cron/lib/cron.js:243
					throw new Error(
					^

Error: Something went wrong. cron reached maximum iterations.
Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
Time Zone: "" - Cron String: 0 0,30 15,16 * * * - UTC offset: -05:00 - current Date: Sat Mar 09 2019 21:35:10 GMT+0000
    at CronTime._getNextDateFrom (/app/node_modules/cron/lib/cron.js:243:12)
    at CronTime.sendAt (/app/node_modules/cron/lib/cron.js:151:17)
    at CronTime.getTimeout (/app/node_modules/cron/lib/cron.js:170:29)
    at CronJob.start (/app/node_modules/cron/lib/cron.js:561:31)
    at new CronJob (/app/node_modules/cron/lib/cron.js:524:10)
    at Object.<anonymous> (/app/server/jobs/jobs.js:10:31)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/app/server/index.js:10:14)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! potluck@0.0.1 prod-server: `node server/index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the potluck@0.0.1 prod-server script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-03-09T21_35_10_098Z-debug.log

**Comment by DavidPatterson-Cole** [comment]
I'm having the same issue. Please let me know if you have found a workaround.

**Comment by nauticalcoder** [comment]
Also experiencing this issue:
Time Zone: "" - Cron String: 0 0 8 * * * - UTC offset: -08:00 - current Date: Sun Mar 10 2019 00:48:02 GMT+0000
It started earlier today.

**Comment by shuat** [comment]
Having the same issue. Just started today.

`Error: Something went wrong. cron reached maximum iterations.
Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
Time Zone: “” - Cron String: 0 11 2 * * * - UTC offset: -05:00 - current Date: Sun Mar 10 2019 01:25:43 GMT+0000
   at CronTime._getNextDateFrom (/app/node_modules/cron/lib/cron.js:243:12)
   at CronTime.sendAt (/app/node_modules/cron/lib/cron.js:151:17)
   at CronTime.getTimeout (/app/node_modules/cron/lib/cron.js:170:29)
   at CronJob.start (/app/node_modules/cron/lib/cron.js:561:31)
   at new CronJob (/app/node_modules/cron/lib/cron.js:524:10)
   at setupCronJobs (/app/utils/cron.js:9:3)
   at swaggerTools.initializeMiddleware (/app/app.js:95:3)`

**Comment by fabnavigator** [comment]
I think this has something to do with daylight savings time. I just changed my time to 1:30a, and the problem went away.

**Comment by nauticalcoder** [comment]
> Also experiencing this issue:
> Time Zone: "" - Cron String: 0 0 8 * * * - UTC offset: -08:00 - current Date: Sun Mar 10 2019 00:48:02 GMT+0000
> It started earlier today.

I changed my string to "0 0 0 * * *" and was able to get my app started again.

**Comment by kcoyner** [comment]
Also having the same issue and also just started early today ....

`
Error: Something went wrong. cron reached maximum iterations.
Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string                                                                
Time Zone: "" - Cron String: 0 0 8 * * * - UTC offset: -05:00 - current Date: Sat Mar 09 2019 21:57:58 GMT-0500
`

**Comment by GaneshaThirumurthi** [comment]
The specific time change is strange. Is it due to daylight savings time?

**Comment by shuat** [comment]
I found the issue
it instantiates a date moment object here https://github.com/kelektiv/node-cron/blob/master/lib/cron.js#L204
when it tries to change the time to 2  over here https://github.com/kelektiv/node-cron/blob/master/lib/cron.js#L298
since that does not exist it stays 1 and gets stuck there

**Comment by ncb000gt** [comment]
Hey. I tried using some of the strings supplied above and it's not acting up for me. It appears to trigger appropriately. I'm not able to reproduce anything any the moment. If you can, I'd definitely review a PR. I'll keep trying to reproduce this though.

**Comment by ncb000gt** [comment]
@shuat landed. thanks for the pr.

**Comment by shuat** [comment]
@ncb000gt I edited my node_modules with this code and it fixed it.

**Comment by ncb000gt** [comment]
@shuat i figured which is why i merged it into master. but, i wanted to make sure that others were seeing this fixed as well.

**Comment by shuat** [comment]
@ncb000gt When will it make it to npm? If it waits until after 2 AM then it won't matter for another year.

**Comment by ncb000gt** [comment]
I just published this to npm as 1.7.0. Thanks @shuat.

**Comment by mryellow** [comment]
> I found the issue
> it instantiates a date moment object here https://github.com/kelektiv/node-cron/blob/master/lib/cron.js#L204
> when it tries to change the time to 2 over here https://github.com/kelektiv/node-cron/blob/master/lib/cron.js#L298
> since that does not exist it stays 1 and gets stuck there

https://github.com/kelektiv/node-cron/blob/5a11d4857655a41dfe4fff9d7f8ccdf10b450a6b/lib/cron.js#L198-L361

This function needs a re-write. So many things to go wrong. They keep going wrong.

**Comment by mikelam14** [comment]
It keeps getting referenced to, so is this fixed or not? I encountered this just today. Is there a solution?

**Comment by mikelam14** [comment]
I realized the error would show whenever my CPU utilization is too high (>9x%). That might cause some delay of code and affect something.

**Comment by mar005** [comment]
Hi I’m getting this issue as well. I saw on this other issue that it said to upgrade to 1.7.0 but that is not the latest docker image from what I saw in docker hub. Is that a dev image?

**Comment by pthor11** [comment]
Same here today


Error: Something went wrong. cron reached maximum iterations.
						Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
						Time Zone: "" - Cron String: 0 * * * * * - UTC offset: +00:00 - current Date:...
  File "May 09 2020 09", line 18, in null.<anonymous>
  File "/usr/app/node_modules/cron/lib/cron.js", line 235, in CronTime._getNextDateFrom
    throw new Error(
  File "/usr/app/node_modules/cron/lib/cron.js", line 156, in CronTime.sendAt
    date = this._getNextDateFrom(date);
  File "/usr/app/node_modules/cron/lib/cron.js", line 175, in CronTime.getTimeout
    return Math.max(-1, this.sendAt() - moment());
  File "/usr/app/node_modules/cron/lib/cron.js", line 613, in CronJob.start
    var timeout = this.cronTime.getTimeout();
...
(3 additional frame(s) were not displayed)

**Comment by fzamperin** [comment]
Same problem

Error: Something went wrong. cron reached maximum iterations.
Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
Time Zone: "" - Cron String: 0,30 * * * * * - UTC offset: +00:00 - current Date: Fri Oct 23 2020 03:39:47 GMT+0000
at CronTime._getNextDateFrom (/home/Apps/packeat-server/node_modules/cron/lib/cron.js:235:12)
at CronTime.sendAt (/home/Apps/packeat-server/node_modules/cron/lib/cron.js:156:17)
at CronTime.getTimeout (/home/Apps/packeat-server/node_modules/cron/lib/cron.js:175:29)
at CronJob.start (/home/Apps/packeat-server/node_modules/cron/lib/cron.js:613:31)
at Timeout.callbackWrapper [as _onTimeout] (/home/Apps/packeat-server/node_modules/cron/lib/cron.js:665:29)
at ontimeout (timers.js:436:11)
at tryOnTimeout (timers.js:300:5)
at listOnTimeout (timers.js:263:5)
at Timer.processTimers (timers.js:223:10)

**Comment by MelodicCrypter** [comment]
I'm also experiencing this one. App is deployed on digitalocean

![Screen Shot 2021-02-03 at 4 27 14 PM](https://user-images.githubusercontent.com/18341500/106723250-df49df80-6641-11eb-97b3-fc55052fb72e.png)

**Comment by koldoon** [comment]
Still getting this error. Ticket should be re-opened

**Comment by adamkdean** [comment]
Experienced this issue with v1.8.2:

`Time Zone: "" - Cron String: 0 * * * * * - UTC offset: +00:00 - current Date: Mon Dec 20 2021 18:10:35 GMT+0000`

**Comment by amrshakya** [comment]
node-cron version: ``v1..8.2``

``Error: Something went wrong. cron reached maximum iterations.
						Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
						Time Zone: "" - Cron String: 0 30 8 * * 1 - UTC offset: +08:00 - current Date: Mon Jan 10 2022 00:42:09 GMT+0000
``

Has there been any fix to this error lately?

**Comment by mryellow** [comment]
The fix is dropping this library. There is a part which needs big refactor and no one working on it.

**Comment by tgt87** [comment]
Having the same issue with v1.8.2 as well:

```
Error: Something went wrong. cron reached maximum iterations.
Please open an issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
Time Zone: "" - Cron String: 0 * * * * * - UTC offset: +08:00 - current Date: Thu Jan 20 2022 06:17:26 GMT+0000
at CronTime._getNextDateFrom (/xxxxx/node_modules/cron/lib/cron.js:235:12)
at CronTime.sendAt (/xxxxx/node_modules/cron/lib/cron.js:156:17)
at CronTime.getTimeout (/xxxxx/node_modules/cron/lib/cron.js:175:29)
at CronJob.start (/xxxxx/node_modules/cron/lib/cron.js:613:31)
at Timeout.callbackWrapper [as _onTimeout] (/xxxxx/node_modules/cron/lib/cron.js:665:29)
at listOnTimeout (internal/timers.js:555:17)
at processTimers (internal/timers.js:498:7)
```

**Comment by Jokurale** [comment]
Hi there, issue still here on version 1.8.2. I can see the fresh one is 3.0.0. Will update solve this one particular occurrence related to moment-timezone lib?

**Comment by nibinkjose** [comment]
0|SSN      | 2022-04-05 11:09:44 +00:00: Error: Something went wrong. cron reached maximum iterations.
0|SSN      | 2022-04-05 11:09:44 +00:00:                                                Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
0|SSN      | 2022-04-05 11:09:44 +00:00:                                                Time Zone: "" - Cron String: 0 * * * * * - UTC offset: +00:00 - current Date: Tue Apr 05 2022 10:56:32 GMT+0000
0|SSN      | 2022-04-05 11:09:44 +00:00:     at CronTime._getNextDateFrom (/home/ubuntu/ssn-api/node_modules/cron/lib/cron.js:235:12)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at CronTime.sendAt (/home/ubuntu/ssn-api/node_modules/cron/lib/cron.js:156:17)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at CronTime.getTimeout (/home/ubuntu/ssn-api/node_modules/cron/lib/cron.js:175:29)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at CronJob.start (/home/ubuntu/ssn-api/node_modules/cron/lib/cron.js:613:31)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at Timeout.callbackWrapper [as _onTimeout] (/home/ubuntu/ssn-api/node_modules/cron/lib/cron.js:665:29)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at ontimeout (timers.js:424:11)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at tryOnTimeout (timers.js:288:5)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at listOnTimeout (timers.js:251:5)
0|SSN      | 2022-04-05 11:09:44 +00:00:     at Timer.processTimers (timers.js:211:10)

**Comment by Hexagon** [comment]
Migrate your application to  https://github.com/hexagon/croner, it won't get stuck in an Infinite loop, and if it does, we will fix it :)

**Comment by intcreator** [comment]
the code to remove Moment.js is already removed, so if that's causing the issue it just needs a publish to NPM to fix. if not I can take a look

**Comment by samw2k00** [comment]
i can see the issue was closed but still having issue with cron 1.8.2

**Comment by sheerlox** [comment]
Hi @samw2k00, this issue has been closed in favor of https://github.com/kelektiv/node-cron/issues/467.
